Framework/Crystal/src/CountReflections.cpp

```diff
@@ -0,0 +1,217 @@
+#include "MantidCrystal/CountReflections.h"
+#include "MantidCrystal/PeakStatisticsTools.h"
+
+#include "MantidAPI/Sample.h"
+#include "MantidAPI/WorkspaceProperty.h"
+
+#include "MantidDataObjects/PeaksWorkspace.h"
+
+#include "MantidGeometry/Crystal/PointGroupFactory.h"
+#include "MantidGeometry/Crystal/ReflectionCondition.h"
+#include "MantidGeometry/Crystal/OrientedLattice.h"
+
+#include "MantidKernel/make_unique.h"
+#include "MantidKernel/ListValidator.h"
+
+namespace Mantid {
+namespace Crystal {
+
+using Mantid::Kernel::Direction;
+
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CountReflections)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string CountReflections::name() const { return "CountReflections"; }
+
+/// Algorithm's version for identification. @see Algorithm::version
+int CountReflections::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string CountReflections::category() const {
+  return "Crystal\\Peaks";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string CountReflections::summary() const {
+  return "Calculates statistics for a PeaksWorkspace based on symmetry and "
+         "counting reflections.";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void CountReflections::init() {
+  declareProperty(Kernel::make_unique<WorkspaceProperty<PeaksWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "A workspace with peaks to calculate statistics for. Sample "
+                  "with valid UB-matrix is required.");
+
+  auto centeringSymbols = getAllReflectionConditionSymbols();
+  declareProperty("LatticeCentering", centeringSymbols[0],
```

Comment:
Yes, that is one thing we should discuss. All other algorithms use the full name, but it seems very verbose and in this context it should be sufficient with the symbol. It's the same for the point group - all other algorithms use the "name", which is the symbol plus the lattice system (if I remember correctly).

"Breaking the rule" here was meant to start that discussion.